### DISCUSSION TITLE
Migrating away from tsc to swc for TypeScript

### DISCUSSION THREAD

**Discussion by repzilon** [swc-project/swc#8754] [Q&A]
Hello,

I have TypeScript projects and I would like to migrate them to use SWC and get rid of this big ugly node_modules directory. However, I have a few questions:

1. How do I migrate my tsconfig file to .swcrc?
2. How can I install SWC **without** npm/yarn/pnpm? SWC is written in Rust, right? So, why would I have to use Node packaging for it?
3. How can I transpile several TypeScript files into a single JavaScript file?
4. I am using Visual Studio Code as my TypeScript editor. As SWC itself does not do type checking, would checking errors with the VSCode-integrated TypeScript checking be enough?

**Comment by kwonoj** [comment]
1. refer https://swc.rs/'s documentation for the corresponding options.
2. There is swc's native cli in beta you can use.
3. you can't.
4. That is something we can't answer for you - every application have different requirements for the level of confidence.
